src/main.rs

```diff
@@ -675,30 +538,139 @@ async fn main() {
             .unwrap_or(172800),
         beaconchain_api_key: env::var("BEACONCHAIN_API_KEY").unwrap_or("".to_owned()),
     });
+    tracing::info!("Config: {shared_state:?}");
 
     let app = Router::new()
         .route("/", get(alive))
-        .route("/info", get(info))
-        .route("/eth/proof/:block_hash", get(get_eth_proof))
-        .route("/eth/head", get(get_eth_head))
-        .route("/avl/head", get(get_avl_head))
-        .route("/avl/proof/:block_hash/:message_id", get(get_avl_proof))
-        .route("/beacon/slot/:slot_number", get(get_beacon_slot))
+        .route("/versions", get(versions))
+        .route("/v1/info", get(info))
+        .route("/v1/eth/proof/:block_hash", get(get_eth_proof))
+        .route("/v1/eth/head", get(get_eth_head))
+        .route("/v1/avl/head", get(get_avl_head))
+        .route("/v1/avl/proof/:block_hash/:message_id", get(get_avl_proof))
         .layer(TraceLayer::new_for_http())
         .layer(CompressionLayer::new())
         .layer(
             CorsLayer::new()
                 .allow_methods(vec![Method::GET])
                 .allow_origin(Any),
         )
-        .with_state(shared_state);
+        .with_state(shared_state.clone());
 
     let host = env::var("HOST").unwrap_or("0.0.0.0".to_owned());
     let port = env::var("PORT").unwrap_or("8080".to_owned());
     let listener = tokio::net::TcpListener::bind(format!("{}:{}", host, port))
         .await
         .unwrap();
 
+    tokio::spawn(async move {
+        tracing::info!("Starting head tracking task");
+        if let Err(e) = track_slot_avail_task(shared_state.clone()).await {
+            tracing::error!("Error occurred, cannot continue: {e:#}");
+            process::exit(-1);
+        }
+    });
     tracing::info!("🚀 Listening on {} port {}", host, port);
     axum::serve(listener, app).await.unwrap();
 }
+
+lazy_static! {
+    static ref SLOT_BLOCK_HEAD: Mutex<Option<(u64, u64, B256, u64)>> = Mutex::new(None);
```

Comment:
Should we use `Arc<RwLock>` here instead?

Reply:
I was thinking about that, right now, it doesn't make a huge difference, but it might in the future. Will change that.

Reply:
Since we already use the `parking_lot` feature in `tokio` might be better to use `parking_lot::RwLock`

Reply:
Shouldn't the tokio::sync::RwLock use parking_lot on the low level? I can also introduce parking_lot separately, but I see no need.